c2corg_api/models/association.py

```diff
@@ -90,33 +95,71 @@ class AssociationLog(Base):
     overrides={})
 
 
-def get_associations(document, lang):
+def get_associations(document, lang, editing_view):
     """Load and return associated documents.
     """
-    def limit_waypoint_fields(query):
-        return query. \
-            options(load_only(
-                Waypoint.waypoint_type, Waypoint.document_id,
-                Waypoint.elevation, Waypoint.version, Waypoint.protected)). \
-            options(joinedload(Waypoint.locales).load_only(
-                WaypointLocale.lang, WaypointLocale.title,
-                WaypointLocale.version))
+    types_to_include = associations_to_include.get(document.type, set())
+
+    if editing_view:
+        edit_types = updatable_associations.get(document.type, set())
+        types_to_include = types_to_include.intersection(edit_types)
+
+    associations = {}
+    if 'waypoints' in types_to_include:
+        waypoint_parents = get_linked_waypoint_parents(document)
+        waypoint_children = get_linked_waypoint_children(document)
+        associations['waypoints'] = waypoint_parents + waypoint_children
+    elif 'waypoint_parents' in types_to_include:
```

Comment:
Not that a big deal but that's a bit not intuitive to get both "waypoint_parents" and "waypoint_children" when ``types_to_include`` only indicates "waypoint_parents" :)